src/litmodels/integrations/mixins.py

```diff
@@ -83,3 +84,81 @@ def pull_from_registry(
         if not isinstance(obj, cls):
             raise RuntimeError(f"Unpickled object is not of type {cls.__name__}: {type(obj)}")
         return obj
+
+
+class PyTorchRegistryMixin(ABC):
+    """Mixin for PyTorch model registry integration."""
+
+    def __post_init__(self) -> None:
+        """Post-initialization method to set up the model."""
+        import torch
+
+        # Ensure that the model is in evaluation mode
+        if not isinstance(self, torch.nn.Module):
+            raise TypeError(f"The model must be a PyTorch `nn.Module` but got: {type(self)}")
+
+    def push_to_registry(
+        self, model_name: Optional[str] = None, model_version: Optional[str] = None, temp_folder: Optional[str] = None
+    ) -> None:
+        """Push the model to the registry.
+
+        Args:
+            model_name: The name of the model. If not use the class name.
+            model_version: The version of the model. If None, the latest version is used.
+            temp_folder: The temporary folder to save the model. If None, a default temporary folder is used.
+        """
+        import torch
+
+        if not isinstance(self, torch.nn.Module):
+            raise TypeError(f"The model must be a PyTorch `nn.Module` but got: {type(self)}")
+
+        if model_name is None:
+            model_name = self.__class__.__name__
+        if temp_folder is None:
+            temp_folder = tempfile.gettempdir()
+        torch_path = Path(temp_folder) / f"{model_name}.pth"
+        torch.save(self.state_dict(), torch_path)
+        # todo: dump also object creation arguments so we can dump it and load with model for object instantiation
+        model_registry = f"{model_name}:{model_version}" if model_version else model_name
+        upload_model(name=model_registry, model=torch_path)
+
+    @classmethod
+    def pull_from_registry(
+        cls,
+        model_name: str,
+        model_version: Optional[str] = None,
+        temp_folder: Optional[str] = None,
+        torch_load_kwargs: Optional[dict] = None,
+    ) -> object:
```

Comment:
shall we make the return type as nn.module here?
```suggestion
    ) -> nn.Module:
```